portal-impl/src/com/liferay/portlet/documentlibrary/service/impl/DLFileEntryLocalServiceImpl.java

```diff
@@ -2771,6 +2790,89 @@ private long _getValidFileEntryTypeId(
 		}
 	}
 
+	private void _inheritRolesPermissions(
+			long parentFolderId, DLFileEntry dlFileEntry)
+		throws PortalException {
+
+		Set<String> commonPermissions = new HashSet<>(
+			Arrays.asList(
+				ActionKeys.DELETE, ActionKeys.PERMISSIONS, ActionKeys.UPDATE,
+				ActionKeys.VIEW));
+
+		long companyId = dlFileEntry.getCompanyId();
+		String parentClassName = DLFolderConstants.getClassName();
+
+		Set<String> dlFolderAllActionIds = SetUtil.fromCollection(
+			ResourceActionsUtil.getModelResourceActions(
+				DLFolderConstants.getClassName()));
+
+		Set<String> dlFileEntryAllActionIds = SetUtil.fromCollection(
```

Comment:
move inside the if statement for better performance

Reply:
I don't think it should be. Because this is used outside